src/UnigramWordPredictor.java

```diff
@@ -52,6 +53,26 @@ public void train(Scanner scanner) {
     List<String> trainingWords = tokenizer.tokenize(scanner);
 
     // TODO: Convert the trainingWords into neighborMap here
+    //initialize map (we in this case cannot do this within the 
+    //constructor, so we can point neighborMap within method to
+    //a new HashMap<>()
+    this.neighborMap = new HashMap<>();
+
+    //iterate over tokens; build NeighborMap
+    for (int i = 0; i < trainingWords.size() - 1; i++) {
+      String word = trainingWords.get(i); 
+      String nextWord = trainingWords.get(i + 1);
+
+      //check if word not in map
+      if (!neighborMap.containsKey(word)) {
+
+      //if not, add word with an empty list as its value
+      neighborMap.put(word, new ArrayList<>());
+      }
```

Comment:
Nice! As an alternative, can you think about how you could write this using `putIfAbsent` or `computeIfAbsent` instead?